test/fmanip/ldexp_test.cpp

```diff
@@ -14,6 +14,60 @@
 
 TEST(CcmathFmanipTests, Ldexp)
 {
+	static_assert(ccm::ldexp(1.0, 0) == 1.0, "ldexp has failed testing that it is static_assert-able!");
 
+	EXPECT_DOUBLE_EQ(std::ldexp(5.0, 4), ccm::ldexp(5.0, 4));
+	EXPECT_DOUBLE_EQ(std::ldexp(5.f, 4), ccm::ldexp(5.f, 4));
+
+	EXPECT_DOUBLE_EQ(std::ldexp(-5.0, -4), ccm::ldexp(-5.0, -4));
+	EXPECT_FLOAT_EQ(std::ldexp(5.f, -4), ccm::ldexp(5.f, -4));
+
+	EXPECT_DOUBLE_EQ(std::ldexp(-0.0, 1), ccm::ldexp(-0.0, 1));
+	EXPECT_FLOAT_EQ(std::ldexp(-0.f, 1), ccm::ldexp(-0.f, 1));
+
+	EXPECT_DOUBLE_EQ(std::ldexp(0.0, 1), ccm::ldexp(0.0, 1));
+	EXPECT_FLOAT_EQ(std::ldexp(0.f, 1), ccm::ldexp(0.f, 1));
+
+	EXPECT_DOUBLE_EQ(std::ldexp(720.32, 22), ccm::ldexp(720.32, 22));
+	EXPECT_FLOAT_EQ(std::ldexp(720.32f, 22), ccm::ldexp(720.32f, 22));
+
+	EXPECT_DOUBLE_EQ(std::ldexp(1, -1074), ccm::ldexp(1, -1074));
+	EXPECT_FLOAT_EQ(std::ldexp(1.f, -1074), ccm::ldexp(1.f, -1074));
+
+	EXPECT_DOUBLE_EQ(std::ldexp(1, 1024), ccm::ldexp(1, 1024));
+	EXPECT_FLOAT_EQ(std::ldexp(1.f, 1024), ccm::ldexp(1.f, 1024));
+
+	EXPECT_DOUBLE_EQ(std::ldexp(std::numeric_limits<double>::max(), std::numeric_limits<int>::max()),
+					ccm::ldexp(std::numeric_limits<double>::max(), std::numeric_limits<int>::max()));
+	EXPECT_FLOAT_EQ(std::ldexp(std::numeric_limits<float>::max(), std::numeric_limits<int>::max()),
+					ccm::ldexp(std::numeric_limits<float>::max(), std::numeric_limits<int>::max()));
+
+	EXPECT_DOUBLE_EQ(std::ldexp(std::numeric_limits<double>::min(), std::numeric_limits<int>::min()),
+					ccm::ldexp(std::numeric_limits<double>::min(), std::numeric_limits<int>::min()));
+	EXPECT_FLOAT_EQ(std::ldexp(std::numeric_limits<float>::min(), std::numeric_limits<int>::min()),
+					ccm::ldexp(std::numeric_limits<float>::min(), std::numeric_limits<int>::min()));
+
+	EXPECT_FLOAT_EQ(std::ldexp(std::numeric_limits<double>::min(), std::numeric_limits<int>::max()),
+					ccm::ldexp(std::numeric_limits<double>::min(), std::numeric_limits<int>::max()));
+	EXPECT_FLOAT_EQ(std::ldexp(std::numeric_limits<float>::min(), std::numeric_limits<int>::max()),
+					ccm::ldexp(std::numeric_limits<float>::min(), std::numeric_limits<int>::max()));
+
+	EXPECT_DOUBLE_EQ(std::ldexp(std::numeric_limits<double>::max(), std::numeric_limits<int>::min()),
+					ccm::ldexp(std::numeric_limits<double>::max(), std::numeric_limits<int>::min()));
+	EXPECT_FLOAT_EQ(std::ldexp(std::numeric_limits<float>::max(), std::numeric_limits<int>::min()),
+					ccm::ldexp(std::numeric_limits<float>::max(), std::numeric_limits<int>::min()));
+
+	EXPECT_DOUBLE_EQ(ccm::ldexp(std::numeric_limits<double>::infinity(), 10), std::ldexp(std::numeric_limits<double>::infinity(), 10));
```

Comment:
We should also be testing for if exp is passed Inf and num is an expected value.